pandas/io/parsers.py

```diff
@@ -656,7 +658,13 @@ def _get_options_with_defaults(self, engine):
         options = {}
 
         for argname, default in compat.iteritems(_parser_defaults):
-            options[argname] = kwds.get(argname, default)
+            value = kwds.get(argname, default)
+
+            if argname == 'mangle_dupe_cols' and value != default:
```

Comment:
I'm confused by this one.  Why don't we want this check to change with `default`?